### PR TITLE
#639: Feature: BITFIELD command with all subcommands

### PR DESCRIPTION
**Pull request by apoorvyadav1111**
Fixes: #639 

Implemented `BITFIELD key [SUBCOMMAND]`. Have handled predominantly most test cases present in the redis tcl tests. implemented unit test cases. 

Additionally, added *.rdb in .gitignore to avoid pushing rdb files to git. 

Design:
Added utility functions in ByteArray struct to handle get set and incrby commands. 

All test cases pass. Once merged #640 can be unblocked.

Thanks
Apoorv

### CODE REVIEW

File: internal/eval/commands.go [hunk 1]
```diff
@@ -926,6 +926,30 @@ var (
 		Arity:    -2,
 		KeySpecs: KeySpecs{BeginIndex: 1},
 	}
+	bitfieldCmdMeta = DiceCmdMeta{
+		Name: "BITFIELD",
+		Info: `The command treats a Redis string as an array of bits,
```
Comment (iamskp11) [inline internal/eval/commands.go:931]: Hey @apoorvyadav1111 , do we need to mention string as "**Redis** string" here?

Comment (apoorvyadav1111) [reply internal/eval/commands.go:931]: Thanks @iamskp11 , I ll fix this.

**Comment by apoorvyadav1111** [comment]
Hi team, @JyotinderSingh , @lucifercr07 , @AshwinKul28 , PR is rebased with the master, Could you please review at your convenience?
Thanks,
Apoorv

**Comment by lucifercr07** [comment]
@apoorvyadav1111 please fix the build once.

**Comment by apoorvyadav1111** [comment]
> @apoorvyadav1111 please fix the build once.

Hi @lucifercr07, thanks for reruning the pipeline. It passed this time. I am trying to recreate the issue in the local meanwhile.
Thanks

File: integration_tests/commands/async/bitfield_test.go [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+	"time"
+
+	testifyAssert "github.com/stretchr/testify/assert"
+)
+
+func TestBitfield(t *testing.T) {
+	conn := getLocalConnection()
+	defer conn.Close()
+
+	defer FireCommand(conn, "FLUSHDB") // clean up after all test cases
```
Comment (JyotinderSingh) [inline integration_tests/commands/async/bitfield_test.go:14]:
```suggestion
        FireCommand(conn, "FLUSHDB")
	defer FireCommand(conn, "FLUSHDB") // clean up after all test cases
```

**Review by JyotinderSingh — APPROVED** [review]
Thanks for the contributions @apoorvyadav1111! The changes look good and well tested!

**Pull request merged**